File: cses/07_mathematics/2208_another-game/solution.py
import sys


def main():
    data = sys.stdin.buffer.read().split()
    idx = 0
    t = int(data[idx]); idx += 1
    out = []
    for _ in range(t):
        n = int(data[idx]); idx += 1
        # Người thứ hai thắng khi và chỉ khi MỌI đống đều chẵn.
        all_even = True
        end = idx + n
        # Duyệt n giá trị; nếu tồn tại một đống lẻ thì người thứ nhất thắng.
        while idx < end:
            if int(data[idx]) & 1:
                all_even = False
                idx = end  # Đã đủ kết luận, bỏ qua phần còn lại của test case này.
                break
            idx += 1
        out.append("second" if all_even else "first")
    sys.stdout.write("\n".join(out) + "\n")


main()

File: cses/07_mathematics/2208_another-game/solution.cpp
#include <iostream>

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int tests;
    std::cin >> tests;
    while (tests-- > 0) {
        int heap_count;
        std::cin >> heap_count;
        // Người thứ hai thắng khi và chỉ khi MỌI đống đều chẵn.
        bool all_even = true;
        for (int heap = 0; heap < heap_count; ++heap) {
            long long coins;
            std::cin >> coins;
            // Chỉ cần một đống lẻ là người thứ nhất thắng.
            if ((coins & 1LL) != 0) {
                all_even = false;
            }
        }
        std::cout << (all_even ? "second" : "first") << '\n';
    }
    return 0;
}
